### PR TITLE
RD2: Data migration dry run

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
Enable users to identify all of the potential issues with their Org and Recurring Donations data before they run the potentially long and irreversible full data migration process so that they may correct any issues with data first. 

# Critical Changes

# Changes

# Issues Closed

# New Metadata
Apex Class:
- RD2_DataMigrationBase_BATCH
- RD2_DataMigrationDryRun_BATCH
- RD2_DataMigrationEnablement
- RD2_DataMigrationEnablement_TEST

Custom Labels:
- RD2_EnablementDryRunBatchTitle
- RD2_EnablementDryRunButton
- RD2_EnablementDryRunComplete
- RD2_EnablementDryRunIntro
- RD2_EnablementDryRunNextButton
- RD2_EnablementDryRunSkipButton
- RD2_EnablementDryRunStopButton
- RD2_EnablementDryRunTitle
- RD2_EnablementMigrationSkipButton
- RD2_EnablementMigrationStepDryRun
- RD2_EnablementMigrationStepFinal
- RD2_EnablementMigrationStopButton

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037553

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037986

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037553

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037553

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HolIQAS" target="_blank">W-037553</a>

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037986

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpaWQAS" target="_blank">W-037986</a>

**Comment by RadSFDeveloper** [comment]
@sam-knox , custom label changes for the dry run, pre-enablement, have been applied. Do you mind please verifying the dry run and migration (step 4) after metadata has been deployed and confirmed? Thank you!

**Comment by sam-knox** [comment]
@RadSFDeveloper - Yes, can do!

**Comment by RadSFDeveloper** [comment]
@Julian88Tex , @sam-knox , @force2b : PR changes have been applied, it is ready for review.

File: src/aura/RD2_EnablementDelegate/RD2_EnablementDelegateHelper.js [hunk 1]
```diff
@@ -165,13 +171,106 @@
...
             this.refreshMetaDeploy(component);
+            this.hideSpinner(component, 'metadeployConfirmSpinner');
+        });
+
+        $A.enqueueAction(action);
+    },
+    /****
+    * @description Starts data migration batch in dry run mode
+    */
+    runDryRun: function (component) {
```
[the remark this answers is not in the document]
Comment (RadSFDeveloper) [reply src/aura/RD2_EnablementDelegate/RD2_EnablementDelegateHelper.js]: The methods seem to diverge now based if it is validation or migration. Please let me know if I overlooked similarities.

File: src/aura/RD2_EnablementDelegate/RD2_EnablementDelegateHelper.js [hunk 2]
```diff
@@ -135,10 +137,10 @@
             const state = response.getState();
 
             if (state === 'SUCCESS') {
-                component.set('v.state.isMetaDeployLaunched', true);
+                component.set('v.state.isMetaLaunched', true);
```
[the remark this answers is not in the document]
Comment (RadSFDeveloper) [reply src/aura/RD2_EnablementDelegate/RD2_EnablementDelegateHelper.js]: Yes, reducing json size is the reason. The flag is used in the LC, but it is still readable. Let me know if I should revert it back.

File: src/labels/CustomLabels.labels [hunk 3]
```diff
@@ -1620,7 +1620,7 @@ If you&apos;re not sure how to resolve these errors, post a message in the Nonpr
         <language>en_US</language>
         <protected>true</protected>
         <shortDescription>RD2_EnablementDryRunIntro</shortDescription>
```
[the remark this answers is not in the document]
Comment (RadSFDeveloper) [reply src/labels/CustomLabels.labels]: Thank you Mike! Custom labels short descriptions have been updated to be descriptive.

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-037812

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HpIRQA0" target="_blank">W-037812</a>

**Comment by Julian88Tex** [comment]
@RadSFDeveloper @npsp-reedestockton RZB does it again! I found the smallest of wrapping issues that I wouldn't consider a blocker to merge that I've notated on a ticket. I completed testing and created a Test Charter. Whenever this ticket is approved, please just let me know any focus areas if there ends up being changes and I expect to knock it out with a small round of re-testing next Monday barring any major changes that need to be made to code.

**Comment by RadSFDeveloper** [comment]
@Julian88Tex , the enablement page buttons have been changes to they are flexible size (changing based on the page width). Thank you for the feedback!

**Comment by RadSFDeveloper** [comment]
Thank you @npsp-reedestockton !
@Julian88Tex , the enablement page has passed the code review.

**Pull request merged**